.github/workflows/ci-ubuntu.yml

```diff
@@ -32,82 +32,53 @@ on:
 ##
 ########################################################################
 
+########################################################################
+## SETTINGS
+##
+## AGDA_BRANCH picks the branch of Agda to use to build the library.
+## It doesn't really track the branch, so you have to drop caches to
+## get a new branch version if it changes. This will be fixed in the
+## next PR.
+########################################################################
+
 env:
-  GHC_VERSION: 8.6.5
-  CABAL_VERSION: 3.2.0.0
+  AGDA_BRANCH: v2.6.3
+  GHC_VERSION: 9.2.5
+  CABAL_VERSION: 3.6.2.0
   CABAL_INSTALL: cabal install --overwrite-policy=always --ghc-options='-O1 +RTS -M6G -RTS'
+  CACHE_PATHS: |
+    ~/.cabal/packages
+    ~/.cabal/store
+    ~/.cabal/bin
 
 jobs:
   test-cubical:
     runs-on: ubuntu-latest
     steps:
-
-      ########################################################################
-      ## SETTINGS
-      ##
-      ## AGDA_BRANCH picks the branch of Agda to use to build the library.
-      ########################################################################
-
-      - name: Initialise variables
-        run: |
-          # Pick Agda version
-          echo "AGDA_BRANCH=v2.6.3" >> $GITHUB_ENV
-
-
-      ########################################################################
-      ## CHECKOUT
-      ########################################################################
-
-      # By default github actions do not pull the repo
-      - name: Checkout cubical
-        uses: actions/checkout@v3
-
-      ########################################################################
-      ## CACHING
-      ########################################################################
+      - name: Install cabal
+        uses: haskell/actions/setup@v2
+        with:
+          ghc-version: ${{ env.GHC_VERSION }}
+          cabal-version: ${{ env.CABAL_VERSION }}
+          cabal-update: true
 
       # This caching step allows us to save a lot of building time by only
       # downloading ghc and cabal, rebuilding Agda, and re-checking unchanged
```

Comment:
Minor nitpick, it would help to remove the "downloading ghc and cabal" part. Not a blocker though IMO.